gnucash/gnome/gnc-plugin-page-invoice.c

```diff
@@ -292,34 +292,6 @@ static GObjectClass *parent_class = NULL;
 /*                      Implementation                      */
 /************************************************************/
 
-GType
-gnc_plugin_page_invoice_get_type (void)
-{
-    static GType gnc_plugin_page_invoice_type = 0;
-
-    if (gnc_plugin_page_invoice_type == 0)
-    {
-        static const GTypeInfo our_info =
-        {
-            sizeof (GncPluginPageInvoiceClass),
-            NULL,
-            NULL,
-            (GClassInitFunc) gnc_plugin_page_invoice_class_init,
-            NULL,
-            NULL,
-            sizeof (GncPluginPageInvoice),
-            0,
-            (GInstanceInitFunc) gnc_plugin_page_invoice_init
-        };
-
-        gnc_plugin_page_invoice_type = g_type_register_static (GNC_TYPE_PLUGIN_PAGE,
-                                       "GncPluginPageInvoice",
-                                       &our_info, 0);
-    }
-
-    return gnc_plugin_page_invoice_type;
-}
-
```

Comment:
Same thing here. Compiles fine but gets stuck in loading. Need to compare the get_type provided by macro with the one in the source. Or maybe the DEFINE_WITH_PRIVATE arguments are wrong, specifically the type.

Reply:
I wonder if there is a macro that doesn't create a get_type function so that the original ones could stay.

Reply:
Umm, this one doesn't appear to need private info so it should just be ```G_DEFINE_TYPE```... but I don't see ```G_DEFINE_TYPE (GncPluginPageInvoice, gnc_plugin_page_invoice, GNC_TYPE_PLUGIN_PAGE)``` in the commit. Not defining it will certainly cause the plugin to not load.

Reply:
It has g_type_class_add_private(klass, sizeof(GncPluginPageInvoicePrivate)); so there is private info. I've been rewriting history by force pushing here.

Reply:
Not loading happens when I remove the existing gnc_plugin_page_invoice_get_type in line 295-321 and use a macro of the G_DEFINE_TYPE family. In this case G_DEFINE_TYPE_WITH_PRIVATE in line 325.
https://github.com/Gnucash/gnucash/pull/439/files/d6f9ccacd67892a54b381a911b833d01f1b5ed97..ea499a5b712870b0c5fd6defcb22443f1de42aac#diff-8bd91e5a24f28088b263673ae92e2946R325